src/monmove.cpp

```diff
@@ -536,6 +536,23 @@ static float get_stagger_adjust( const tripoint &source, const tripoint &destina
     return std::max( 0.01f, initial_dist - new_dist );
 }
 
+bool monster::die_if_drowning( const tripoint &at_pos, const int chance )
+{
+    if( g->m.has_flag( "LIQUID", at_pos ) && can_drown() && one_in( chance ) ) {
+        vehicle *veh = veh_pointer_or_null( g->m.veh_at( at_pos ) );
+        // if there's a vehicle here, the monster is on it and not drowning
```

Comment:
This should check for boardable parts. A zombie standing on a wing mirror would like interesting, but it's not even close to being realistic.

Reply:
Fixed, thanks.